eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Epoch.hs

```diff
@@ -266,7 +267,7 @@ epochTransition = do
           & prevPParamsEpochStateL .~ pp
           & curPParamsEpochStateL .~ pp
       -- TODO can we be more efficient?
-      newGov = GovActionsState . Map.fromList . toList $ rsFuture
+      newGov = foldr' insertGovActionEntry mempty rsFuture
```

Comment:
Is it this in particular that improves the strictness in this solution?

Reply:
The main improvement is that the fields in `GovActionEntry` are strict as opposed to the pair we had before. Here I managed to avoid a conversion to list which is also nice.

Reply:
Makes sense, thank you for explaining.

Reply:
@Soupstraw should we remove the TODO now that you have make it efficient enough? Or does it still remain?